File: SSI/filters/exponential_decay.py
def exponential_decay(input_data, alpha):
    '''
    Exponential Decay Smoothing Filter
    Increase alpha for more filtering: uses more past data to compute an average
    - input_data is a vector (list of values)
    '''

    output_data = input_data.copy() 
    for i in range(0,input_data.num-1):
        output_data[i+1] = (output_data[i,0]*alpha) + (output_data[i+1,0] * (1-alpha))
        
    return output_data

File: Hub Location Method Analysis/library/BIAPT-NeuroAlgo-0c6969d/MATLAB/filters/exponential_decay.m
function [filtered_data] = exponential_decay(data, alpha)
    %Exponential decay smoothing filter
    %increase alpha for more filtering: uses more past data to compute an
    %average
    filtered_data = data;
    for i = 1:(length(data)-1)
        filtered_data(i+1) = (filtered_data(i)*alpha) + (data(i+1)*(1-alpha));
    end 
end

